File: tutorial/axemas-ios/BaseSectionController.m
//
//  BaseSectionController.m
//  Tutorial
//
//  Created by Andrei Neagu on 15/10/14.
//  Copyright (c) 2014 AXANT. All rights reserved.
//

#import "BaseSectionController.h"
#import "NavigationSectionsManager.h"
#import "ExampleViewController.h"

@implementation BaseSectionController

-(void)sectionWillLoad{
    [super sectionWillLoad];
    
    [self.section.bridge registerHandler:@"push-native-view" handler:^(id data, WVJBResponseCallback responseCallback) {
        NSDictionary *receivedData = (NSDictionary*)data;
        NSNumber * close = (NSNumber *)[receivedData objectForKey: @"close"];
        if([close boolValue] == YES )
            [[NavigationSectionsManager activeSidebarController] revealToggleAnimated:YES];
        [NavigationSectionsManager pushController:[[ExampleViewController alloc] init] animated:YES];
        
        if (responseCallback) {
            responseCallback(nil);
        }
    }];
    
}

@end

File: tutorial/axemas-android/app/src/main/java/it/axant/tutorial/BaseSectionController.java
package it.axant.tutorial;

import org.json.JSONException;
import org.json.JSONObject;

import it.axant.axemas.AXMSectionController;
import it.axant.axemas.JavascriptBridge;
import it.axant.axemas.NavigationSectionsManager;
import it.axant.axemas.SectionFragment;


public class BaseSectionController extends AXMSectionController {

    public BaseSectionController(SectionFragment section) {
        super(section);
    }

    @Override
    public void sectionWillLoad() {
        super.sectionWillLoad();

        this.section.getJSBridge().registerHandler("push-native-view", new JavascriptBridge.Handler() {
            @Override
            public void call(Object data, JavascriptBridge.Callback callback) {
                JSONObject receivedData = (JSONObject) data;
                try {
                    if(receivedData.getBoolean("close")== true)
                        NavigationSectionsManager.toggleSidebar(section.getActivity(),false);
                    NavigationSectionsManager.pushFragment(section.getActivity(), new ExampleFragment());
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        });
    }
}
